### ISSUE TITLE
Doesn't launch over OpenSSH session

### ISSUE THREAD

**Issue by DCH3416** [samizzo/hexed#5]
When connected over ssh using [OpenSSH server on Windows](https://learn.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse). Hexed launches and then immediately quits. 

This includes when launching with `hexed.exe filename` argument.

**Comment by samizzo** [comment]
Thanks for the bug report! It might be that it's not possible to support it since I'm not sure how well (or at all) the Windows console API works over SSH. I'll take a look.

**Comment by DCH3416** [comment]
> Thanks for the bug report! It might be that it's not possible to support it since I'm not sure how well (or at all) the Windows console API works over SSH. I'll take a look.

Thanks. Hope it's not too complicated.

**Comment by samizzo** [comment]
I installed OpenSSH on my PC (Windows 10) and sshed to it (so just ssh localhost). It definitely doesn't run without arguments because in that case it pops up an open file dialog, so that makes sense (I might be able to detect if that can't be shown and print out a usage message instead).

I'm seeing super inconsistent behaviour when you pass in a filename though. The first time I tried it behaved as you mentioned - just quit immediately. So I added in a `printf` in `main` and it suddenly started working. Removed it, and it was still working! Restarted terminal, tried in cmd, powershell, and Windows Terminal using cmd, and it was working in all cases. Very strange. Still investigating.

**Comment by DCH3416** [comment]
I tested dropping a printf in main, and I took out the GUI open dialog. Recompiled using MinGW GCC 8.1.0, and it started working over SSH. 

The OpenSSH server seems to support at launching simple win32 GUI applications. I've tested it using the classic calc app program. Although you can't really interact with what you launch.

**Comment by samizzo** [comment]
I set up a free Azure account and a Windows Server 2022 Datacentre Azure Edition VM with OpenSSH. I SSHed in and downloaded the latest release that I've just put up on github. The first thing I got was missing DLLs because it didn't seem to have the redistributables installed so I installed [those](https://learn.microsoft.com/en-US/cpp/windows/latest-supported-vc-redist?view=msvc-170). I was a bit surprised it didn't have them. I'm building with VS2022 now incidentally.

After that this latest release seems to be working. Unfortunately I couldn't reproduce the issue you originally reported or the weird flakiness I had either. I have fixed it so that it doesn't pop up the open file dialog if it detects that it's running under SSH at least so I guess I'll close this issue.

**Comment by DCH3416** [comment]
I downloaded the latest release. SSH'd and ran it on my desktop PC, and it worked. 

I then copied it onto the other machine that I started this thread with. Ran it, and it didn't. Then I recompiled it, ran that, and that worked. So. It may be a machine specific problem, maybe some absent library, or something so intermittent with windows. But yeah. It sounds like for the most part it's behaving normally. 

Heh, not including DLLs sounds very Microsoft.
